ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/engine/AbstractEngineForkchoiceUpdated.java

```diff
@@ -93,37 +93,26 @@ public JsonRpcResponse syncResponse(final JsonRpcRequestContext requestContext)
             forkChoice.getSafeBlockHash(),
             forkChoice.getFinalizedBlockHash());
 
-    if (mergeCoordinator.isBadBlock(forkChoice.getHeadBlockHash())) {
```

Comment:
This will break the Invalid Missing Ancestor ReOrg - Reveal using sync hive tests that were fixed in this https://github.com/hyperledger/besu/pull/5967.

It happens when the invalid ancestor is in the bad blocks but we keep returning in sync.

Reply:
it's still there, just got moved down to line 148

Reply:
the problem is that line 101 happens before it